src/widget_type.h

```diff
@@ -105,6 +108,14 @@ enum WidgetType {
 	NWID_PUSHBUTTON_DROPDOWN = NWID_BUTTON_DROPDOWN | WWB_PUSHBUTTON,
 };
 
+/**
+ * Base widget flags.
+ */
+enum WidgetBaseFlags : uint8 {
+	WBF_DIRTY            = 1 <<  0, ///< Widget is dirty.
+};
+DECLARE_ENUM_AS_BIT_SET(WidgetBaseFlags)
```

Comment:
Would it be worth extending this macro to include `+`, `-`, `+=`, and `-=` operators for setting and clearing individual bits (in a separate PR, of course)? There must be plenty of opportunity in the codebase to use them.